src/script/ismine.h

```diff
@@ -13,6 +13,8 @@
 class CKeyStore;
 class CScript;
 
+static const bool DEFAULT_ANYONECANSPEND_AREMINE = true;
```

Comment:
imo this should be false by default. It's a hack for regtest, for the most part.

Reply:
Mhmm, my thought was that custom/regtest is the only chain that we support directly...perhaps it should be a field in CChainParams instead?
The important part is removing ```Params().NetworkIDString() == CHAINPARAMS_REGTEST```